### PR TITLE
Medical - Add functions to serialize / deserialize the medical state of a unit

### PR DESCRIPTION
**Pull request by BaerMitUmlaut**
**When merged this pull request will:**
- Add functions to serialize / deserialize the medical state of a unit to / from JSON

I considered just returning the CBA namespace, but I think a JSON string is easier to handle for DBs and such.

### CODE REVIEW

File: addons/medical/functions/fnc_deserializeState.sqf [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+ * Public: Yes
+ */
+params [["_unit", objNull, [objNull]], ["_json", "{}", [""]]];
+
+if (isNull _unit) exitWith {};
+if (!local _unit) exitWith { ERROR_1("unit [%1] is not local",_unit) };
+
+// If unit is not initialized yet, wait until event is raised
+if !(_unit getVariable [QGVAR(initialized), false]) exitWith {
+    [QGVAR(initialized), {
```
Comment (tbeswick96) [inline addons/medical/functions/fnc_deserializeState.sqf:25]: Shouldn't this event name be `QEGVAR(medical_status,initialized)`?

Comment (jonpas) [reply addons/medical/functions/fnc_deserializeState.sqf:25]: @BaerMitUmlaut ?

**Comment by BrettMayson** [comment]
Status on this? Anything blocking?

File: addons/medical/functions/fnc_deserializeState.sqf (hunk 1 above)
Comment (BaerMitUmlaut) [reply addons/medical/functions/fnc_deserializeState.sqf]: Fixed.

**Comment by BaerMitUmlaut** [comment]
Should be good to go now.

**Review by jonpas — APPROVED** [review]

**Pull request merged**